caom2-search-server/src/main/java/ca/nrc/cadc/search/PackageServlet.java

```diff
@@ -119,16 +120,34 @@ protected void doGet(final HttpServletRequest request, final HttpServletResponse
      */
     void get(final HttpServletRequest request, final HttpServletResponse response, final RegistryClient registryClient)
         throws IOException, URISyntaxException {
-        final URL serviceURL = registryClient.getServiceURL(getServiceID(CAOM2PKG_SERVICE_URI_PROPERTY_KEY,
-                                                                         DEFAULT_CAOM2PKG_SERVICE_URI),
-                                                            Standards.PKG_10, AuthMethod.COOKIE);
 
-        final URIBuilder builder = new URIBuilder(serviceURL.toURI());
+        // TODO: prior to version 2.5, this servlet supported multiple IDs.
+        // Consider how this might be supported in future.
+        String[] idValues = request.getParameterValues("ID");
+        if (idValues.length > 1) {
+            throw new UnsupportedOperationException("Multiple IDs in package lookup.");
+        }
+
+        String IDValue = idValues[0];
+        if (IDValue.length() > 0 ) {
+            // split the ID parameter on '?' to pull off the query string.
+            // use the first part (resourceid) in the getServiceURL
+            String[] uri_parts = IDValue.split("\\?");
```

Comment:
There is a class called PublisherID in the caom2 library that will parse the URI and extract the resourceID for you...